common/src/main/java/org/apache/iceberg/common/DynMethods.java

```diff
@@ -163,10 +162,6 @@ private BoundMethod(UnboundMethod method, Object receiver) {
       this.receiver = receiver;
     }
 
-    public <R> R invokeChecked(Object... args) throws Exception {
```

Comment:
Thought to removed these unused methods after deprecating invokeChecked() . PLMK if it breaks any changes.

Reply:
this method wasn't marked as deprecated, so we can't remove it